File: python/7 kyu/369.py
# https://www.codewars.com/kata/5aee86c5783bb432cd000018/train/python

# Passed

def hydrate(drink_string): 
    all_numbers = [int(num) for num in drink_string.split() if num.isdigit()]
    total_sum = sum(all_numbers)
    
    result = f"{total_sum} {'glass' if total_sum == 1 else 'glasses'} of water"
    return result

output = hydrate("1 shot, 5 beers, 2 shots, 1 glass of wine, 1 beer")
print(output)

File: nodejs/7 kyu/369.js
// https://www.codewars.com/kata/559e10e2e162b69f750000b4/train/javascript

// Passed

function dominator(arr) {
  let valueToCountMap = new Object();
  for (const num of arr) {
    if (!(num in valueToCountMap)) {
      valueToCountMap[num] = 0;
    }

    valueToCountMap[num]++;
  }

  let result = -1;
  let lengthToCompare = Math.floor(arr.length / 2);
  for (const num of Object.keys(valueToCountMap)) {
    if (valueToCountMap[num] > lengthToCompare) {
      result = Number(num);
      break;
    }
  }

  return result;
}

const output = dominator([3, 4, 3, 2, 3, 1, 3, 3]);
console.log(output);